rcl/addtwoints_client/main.c

```diff
@@ -5,105 +5,74 @@
 #include <stdio.h>
 #include <unistd.h>
 
+#define RCCHECK(fn) rc = fn; if((rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Aborting.\n",__LINE__,(int)rc); return 1;}
```

Comment:
Just as an idea...
```suggestion
#define RCCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Aborting.\n",__LINE__,(int)temp_rc); return 1;}}
```
In that way you do not depende on the `rc` variable.

Reply:
Done